src/executor/sample_executor.py

```diff
@@ -0,0 +1,49 @@
+# coding=utf-8
+# Copyright 2018-2020 EVA
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Iterator
+
+from src.models.storage.batch import Batch
+from src.executor.abstract_executor import AbstractExecutor
+from src.planner.sample_plan import SamplePlan
+from src.configuration.configuration_manager import ConfigurationManager
+
+
+class SampleExecutor(AbstractExecutor):
+    """
+    Samples uniformly from the rows.
+
+    Arguments:
+        node (AbstractPlan): The Sample Plan
+
+    """
+
+    def __init__(self, node: SamplePlan):
+        super().__init__(node)
+        self._sample_freq = node.sample_value
+        self.BATCH_MAX_SIZE = ConfigurationManager().get_value(
+            "executor", "batch_size")
+
+    def validate(self):
+        pass
+
+    def exec(self) -> Iterator[Batch]:
+        child_executor = self.children[0]
+        # aggregates the batches into one large batch
+        current = 0
+        for batch in child_executor.exec():
+            start = current
+            for i in range(start, len(batch), self._sample_freq):
+                yield batch[i:i + 1]
```

Comment:
This will make the batch size become one after the sample operator.  Maybe using `df[df.index.isin([1,3])]` to select dataframe rows by a list of index is better. df is the panda dataframe object. [1,3] is the desired row indices.

Reply:
Actually better to use _get_frames_from_indices from the Batch class.

Reply:
@gaurav274 we might want to change the _get_frames_from_indices's implementation check
https://stackoverflow.com/questions/19155718/select-pandas-rows-based-on-list-index

Reply:
Fixed.

Reply:
Sorry, I should point to the slicing support for Batch (which internally calls the _get_frames_from_indices). So 
```
filter = range(current, len(batch), self._sample_freq)
yield batch._get_frames_from_indices(filter)
```
can be simplified to `Batch[current::self._sample_freq]`.

Reply:
I don't know if that makes sense because I need to keep changing the start index. I cant start at zero.

Reply:
It's a slicing representation [start:stop:step], so Batch[current::self._sample_freq] start at current, instead of zero.

Reply:
Fixed.